lib/static_site_generator.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'capybara-webkit'
+require 'pathname'
+require 'uri'
+
+class StaticSiteGenerator
+  attr_reader :page, :urls
+
+  def initialize(urls:)
+    Capybara::Webkit.configure(&:allow_unknown_urls)
+    @page = Capybara::Session.new(:webkit)
+    @urls = urls.map { |u| URI.parse(u) }
```

Comment:
NB: our practice in this repo is to avoid doing work directly in the constructor — most of these seem like things that can be deferred to the methods that ask for them.